packages/hapi/modules/plugin/plugin.ts

```diff
@@ -23,9 +24,19 @@ interface TPromsterRequest extends Request {
 }
 
 const extractPath = (request: Request) => request.route.path.replace(/\?/g, '');
-const extractStatusCode = (request: Request) =>
-  // @ts-expect-error
-  request.response ? request.response.statusCode : '';
+const extractStatusCode = (request: Request) => {
+  const { response } = request;
+
+  if (!response) {
+    return '';
+  }
+
+  if ((response as Boom).isBoom) {
```

Comment:
Maybe we can use TypeScript [type guards](https://www.typescriptlang.org/docs/handbook/advanced-types.html#user-defined-type-guards) for this.

```js
type TResponse = Boom | ResponseObject;

const isBoomResponse => (response: TResponse): x is Boom => response.isBoom
const isResponse => (response: TResponse): x is ResponseObject => !response.isBoom

const response: TResponse = request.response;

if (!response) return ''

if (isBoomResponse(response) return response.output.statusCode
if (isResponse(response) return response.statusCode

return ''
```

It would avoid the casting and lift things more into TypeScript's typesystem. What do you think?

Reply:
Yeah, I'm into that. I'll make the changes.

Reply:
@tdeekens Just pushed up these changes. Figured that only `isBoomResponse` was strictly necessary here. Happy to add `isResponse` if we truly need it.

Reply:
Yeah you‘re right. Went a bit over board with type guards ;)